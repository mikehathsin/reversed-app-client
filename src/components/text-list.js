import Badge from "react-bootstrap/Badge";
import Stack from "react-bootstrap/Stack";

export const TextList = ({ data }) => {
  const mapDataToElements = () => {
    return data.map(({ id, text, palindrome, datetime }) => {
      return (
        <div key={id} className="p-2 d-flex flex-column bg-light border">
          <div className="d-flex justify-content-between">
            <span>{text}</span>
            {palindrome && <Badge bg="success">It's palindrome!</Badge>}
          </div>
          <div className="align-self-end">{datetime}</div>
        </div>
      );
    });
  };

  return (
    <Stack gap={2}>
      {Boolean(data.length) ? (
        mapDataToElements()
      ) : (
        <h4>No data. Send your first text!</h4>
      )}
    </Stack>
  );
};
