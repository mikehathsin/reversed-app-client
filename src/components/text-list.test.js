import { render } from "@testing-library/react";

import { TextList } from "./text-list";

describe("TextList tests", () => {
  it("Should render correctly without data", () => {
    const component = render(<TextList data={[]} />);
    component.getByText("No data. Send your first text!");
    expect(component.container).toMatchSnapshot();
  });

  it("Should render correctly with data", () => {
    const component = render(
      <TextList
        data={[
          { id: "1", text: "ana", palindrome: true, datetime: "2:23:43 PM" },
        ]}
      />
    );
    component.getByText("ana");
    component.getByText("It's palindrome!");
    component.getByText("2:23:43 PM");
    expect(component.container).toMatchSnapshot();
  });
});
