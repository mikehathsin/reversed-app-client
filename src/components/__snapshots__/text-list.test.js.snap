// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextList tests Should render correctly with data 1`] = `
<div>
  <div
    class="vstack gap-2"
  >
    <div
      class="p-2 d-flex flex-column bg-light border"
    >
      <div
        class="d-flex justify-content-between"
      >
        <span>
          ana
        </span>
        <span
          class="badge bg-success"
        >
          It's palindrome!
        </span>
      </div>
      <div
        class="align-self-end"
      >
        2:23:43 PM
      </div>
    </div>
  </div>
</div>
`;

exports[`TextList tests Should render correctly without data 1`] = `
<div>
  <div
    class="vstack gap-2"
  >
    <h4>
      No data. Send your first text!
    </h4>
  </div>
</div>
`;
